python/tidy/linting_report.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2025 The Servo Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+from dataclasses import dataclass
+from typing import Any, Literal, NotRequired
+
+
+@dataclass
+class GithubAnnotation:
+    file_name: str
+    line_start: int
+    line_end: int
+    level: Literal["notice", "warning", "error"]
+    title: str
+    message: str
+    column_start: NotRequired[int]
+    column_end: NotRequired[int]
+
+
+class GitHubAnnotationManager:
+    def __init__(self, annotation_prefix: str, limit: int = 10):
+        self.annotation_prefix: str = annotation_prefix
+        self.limit: int = limit
+        self.severenty_map: dict[str, Literal["notice", "warning", "error"]] = {
+            "help": "notice",
+            "note": "notice",
+            "warning": "warning",
+            "error": "error",
+        }
+        self.total_count: int = 0
+
+    def clean_path(self, path: str):
+        return path.removeprefix("./")
+
+    def escape(self, s: str):
+        return s.replace("\r", "%0D").replace("\n", "%0A")
+
+    def emit_annotation(
+        self,
+        title: str,
+        message: str,
+        file_name: str,
+        line_start: int,
+        line_end: int = None,
+        annotation_level: str = None,
+        column_start: int = None,
+        column_end: int = None,
```

Comment:
I think the types of these need to be explicitly also `None`:
```suggestion
        column_start: int | None = None,
        column_end: int | None = None,
```